server/jwt_test.go

```diff
@@ -6787,3 +6787,134 @@ func TestJWTAccountConnzAccessAfterClaimUpdate(t *testing.T) {
 	// If export was wiped this would fail with timeout.
 	doRequest()
 }
+
+func TestJWTSysAccUpdateMixedMode(t *testing.T) {
+	skp, spub := createKey(t)
+	sUsr := createUserCreds(t, nil, skp)
+	sysClaim := jwt.NewAccountClaims(spub)
+	sysClaim.Name = "SYS"
+	sysJwt := encodeClaim(t, sysClaim, spub)
+	encodeJwt1Time := time.Now()
+
+	akp, apub := createKey(t)
+	aUsr := createUserCreds(t, nil, akp)
+	claim := jwt.NewAccountClaims(apub)
+	claim.Limits.JetStreamLimits.DiskStorage = 1024 * 1024
+	claim.Limits.JetStreamLimits.Streams = 1
+	jwt1 := encodeClaim(t, claim, apub)
+
+	basePath := "/ngs/v1/accounts/jwt/"
+	reqCount := int32(0)
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path == basePath {
+			w.Write([]byte("ok"))
+		} else if strings.HasSuffix(r.URL.Path, spub) {
+			w.Write([]byte(sysJwt))
+		} else if strings.HasSuffix(r.URL.Path, apub) {
+			w.Write([]byte(jwt1))
+		} else {
+			// only count requests that could be filled
+			return
+		}
+		atomic.AddInt32(&reqCount, 1)
+	}))
+	defer ts.Close()
+
+	tmpl := `
+		listen: 127.0.0.1:-1
+		server_name: %s
+		jetstream: {max_mem_store: 256MB, max_file_store: 2GB, store_dir: '%s'}
+		cluster {
+			name: %s
+			listen: 127.0.0.1:%d
+			routes = [%s]
+		}
+    `
+
+	sc := createJetStreamSuperClusterWithTemplateAndModHook(t, tmpl, 3, 2,
+		func(serverName, clusterName, storeDir, conf string) string {
+			// create an ngs like setup, with connection and non connection server
+			if clusterName == "C1" {
+				conf = strings.ReplaceAll(conf, "jetstream", "#jetstream")
+			}
+			return fmt.Sprintf(`%s
+				operator: %s
+				system_account: %s
+				resolver: URL("%s%s")`, conf, ojwt, spub, ts.URL, basePath)
+		})
+	defer sc.shutdown()
+	disconnectChan := make(chan struct{}, 100)
+	defer close(disconnectChan)
+	disconnectCb := nats.DisconnectErrHandler(func(conn *nats.Conn, err error) {
+		disconnectChan <- struct{}{}
+	})
+
+	s := sc.clusterForName("C1").randomServer()
+
+	sysNc := natsConnect(t, s.ClientURL(), sUsr, disconnectCb)
+	defer sysNc.Close()
+	defer sysNc.SetDisconnectErrHandler(nil)
```

Comment:
FYI: Instead of doing that, you could pass the option `nats.NoCallbacksAfterClientClose()` in the Connect() call. With that option, the disconnect (and close) handler(s) won't be invoked in Close()

Reply:
And you don't need to close the disconnectChan, regardless. You are using a buffered channel, so that won't block the disconnect handler on exit, even if you were not using NoCallbacksAfterClientClose option.

Reply:
switched to NoCallbacksAfterClientClose instead of clearing the error handler. thanks for that tip.
I'll keep the channel close, seems off without.